Library/Homebrew/cask/cmd/upgrade.rb

```diff
@@ -93,6 +93,12 @@ def self.upgrade_casks(
           end
         end
 
+        manual_installer_casks = outdated_casks.select do |cask|
+          cask.artifacts.any?(Artifact::Installer::ManualInstaller)
+        end
+
+        outdated_casks -= manual_installer_casks
```

Comment:
It may be worth messaging that these aren't upgraded (like we do for pinned formulae). What do you think?

Reply:
> It may be worth messaging that these aren't upgraded (like we do for pinned formulae). What do you think?

I agree with you,  I will add the message like `Not upgrading ~ casks because they were installed manually`